lavalink/src/node.rs

```diff
@@ -488,6 +498,15 @@ impl Connection {
     }
 }
 
+impl Drop for Connection {
+    fn drop(&mut self) {
+        // Cleanup local players associated with the node
+        self.players
+            .players
+            .retain(|_, v| v.node().config() != &self.config);
+    }
+}
```

Comment:
Could this course long stalls?

Reply:
IIRC, in DashMap, when running `retain`, each shard is independently locked, so even if there are a lot of active players it won't block access to the entire map while iterating. This also doesn't have the cost of repeatedly acquiring and releasing the locks on each shard (unlike a loop calling `remove`).

As for the efficiency of comparing configs against each other, if there is a better way to identify which player belongs to which node/connection, I'm all ears.